### PR TITLE
feat: support sending alarm to discord

### PR DESCRIPTION
**Pull request by BFergerson**
Modeled sending alarms to Discord after the WeLink webhook

Screenshot:
image

### CODE REVIEW

File: oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/discord/DiscordHookCallback.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    /**
+     * Send alarm message if the settings not empty
+     */
+    @Override
+    public void doAlarm(List<AlarmMessage> alarmMessages) {
+        if (this.alarmRulesWatcher.getDiscordSettings() == null || this.alarmRulesWatcher.getDiscordSettings()
+                .getWebhooks()
+                .isEmpty()) {
```
Comment (wu-sheng) [inline oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/discord/DiscordHookCallback.java:71]: These lines seem formatted strange, could you make the second condition as a new line totally?

File: oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/discord/DiscordHookCallback.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            return;
+        }
+        DiscordSettings discordSettings = this.alarmRulesWatcher.getDiscordSettings();
+        discordSettings.getWebhooks().forEach(webHookUrl -> {
+            alarmMessages.forEach(alarmMessage -> {
+                String content = String.format(
+                        Locale.US,
```
Comment (wu-sheng) [inline oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/discord/DiscordHookCallback.java:78]: Does US relate to language or encoding or something? Should we move this to configuration?

File: oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/discord/DiscordHookCallback.java (hunk 1 above)
Comment (BFergerson) [reply oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/discord/DiscordHookCallback.java]: Reformatted

File: oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/discord/DiscordHookCallback.java (hunk 2 above)
Comment (BFergerson) [reply oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/discord/DiscordHookCallback.java]: I copied it from WeLink. I tested that it's not necessary and saw that Dingtalk does not use it. Removed locale.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**